pyfar/dsp/filter/gammatone.py

```diff
@@ -0,0 +1,581 @@
+        # Eq. (1) in Hohmann 2002 (these are the a_1 coefficients)
+        coefficients = lam * np.exp(1j * beta)
+        # normalization from Sec. 2.2 in Hohmann 2002
+        # (this is the b_0 coefficient)
+        normalizations = 2 * (1-np.abs(coefficients))**4
+
+        return coefficients, normalizations
+
+    def _get_delays_and_phase_factors(self):
+        """
+        Section 4 in Hohmann 2002 describes how to derive these values. This
+        is a direct Python port of the corresponding function in the AMT
+        toolbox `hohmann2002_process.m`.
+        """
+
+        # the delay in samples
+        delay_samples = int(np.round(self.delay * self.sampling_rate))
+
+        # apply filterbank to impulse to estimate the required values
+        real, imag = self.filter(pf.signals.impulse(delay_samples + 3))
+
+        # compute the envelope
+        ir = real.time + 1j * imag.time
+        env = np.abs(ir)
+
+        # sample at which the maximum occurs
+        # (excluding last sample for a safe calculation of the slope below)
+        idx_max = np.argmax(env[:, :delay_samples + 1], axis=-1)
+        delays = delay_samples - idx_max
+
+        # calculate the phase factor from the slopes
+        slopes = np.array([ir[bb, idx + 1] - ir[bb, idx - 1]
+                           for bb, idx in enumerate(idx_max)])
+
+        phase_factors = 1j / (slopes / np.abs(slopes))
+
+        return delays, phase_factors
+
+    def _get_gains(self):
+        """
+        Section 4 in Hohmann 2002 describes how to derive these values. This
+        is a direct Python port of the corresponding function in the AMT
+        toolbox `hohmann2002_process.m`.
+        """
+
+        # positive and negative center frequencies in the z-plane
+        z = np.atleast_2d(
+            np.exp(2j * np.pi * self.frequencies / self.sampling_rate)).T
+        z_conj = np.conjugate(z)
+
+        # calculate transfer function at all center frequencies for all bands
+        # (matrixes contain center frequencies along first dimension and
+        h_pos = (1 - np.atleast_2d(self._coefficients) / z)**(-4) * \
+            np.atleast_2d(self._normalizations)
+        h_neg = (1 - np.atleast_2d(self._coefficients) / z_conj)**(-4) * \
+            np.atleast_2d(self._normalizations)
+
+        # apply delay and phase correction
+        phase_factors = np.atleast_2d(self._phase_factors)
+        delays = np.atleast_2d(self._delays)
+        h_pos *= phase_factors * z**(-delays)
+        h_neg *= phase_factors * np.conjugate(z)**(-delays)
+
+        # combine positive and negative spectrum
+        h = (h_pos + np.conjugate(h_neg)) / 2
+
+        # iteratively find gains
+        gains = np.ones((self.n_bands, 1))
+        for ii in range(100):
+            h_fin = np.matmul(h, gains)
+            gains /= np.abs(h_fin)
+
+        return gains.flatten()
+
+    def filter(self, signal, reset_state=True):
```

Comment:
This is inconsistent with our other filter classes.

```suggestion
    def process(self, signal, reset=True):
```

Reply:
done as suggested